### ISSUE TITLE
SAM does not support empty SemanticVersion in AWS::Serverless::Application resource

### ISSUE THREAD

**Issue by ylynn** [awslabs/serverless-application-model#891]
**Description:**

SAM does not support empty SemanticVersion in AWS::Serverless::Application resource.

Using template:

```
Resources:
    helloworld:
      Type: AWS::Serverless::Application
      Properties:
        Location:
          ApplicationId: arn:aws:serverlessrepo:us-east-1:077246666028:applications/hello-world
          SemanticVersion:
        Parameters:
          IdentityNameParameter: YOUR_VALUE
```

Fail with an unhandled exception.

**Comment by keetonian** [comment]
Fix for this issue:

1. Validate that parameters (ApplicationId and SemanticVersion) are not `None` in https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/plugins/application/serverless_app_plugin.py#L126 if these properties exist in `Location`.
1. Add an error-case test in https://github.com/awslabs/serverless-application-model/blob/develop/tests/translator/test_translator.py#L444

**Comment by jadhavmanoj** [comment]
@keetonian I ran test case for this test case. but seems it already handled the exception. I can see the error message as "Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [helloworld] is invalid. Property 'SemanticVersion' cannot be blank."

If I understand correctly, It should not allow (ApplicationId and SemanticVersion)  as None. correct? 

I added this PR with a test case for the semantic version as None/empty.
